### PR TITLE
Convert Mesh to TypeScript

### PR DESCRIPTION
**Pull request by Zyie**
##### Description of change

Converts Mesh to TypeScript. Nice and simple
##### Pre-Merge Checklist

- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: packages/mesh/src/Mesh.ts [hunk 1]
```diff
@@ -399,7 +427,7 @@ export class Mesh extends Container
         }
         else
         {
-            this.uvs = geomUvs.data;
+            this.uvs = geomUvs.data as Float32Array;
```
Comment (ivanpopelyshev) [inline packages/mesh/src/Mesh.ts]: We can use `ITypedArray` in such places in future, but its fine to convert it if we know certain type.

File: packages/mesh/src/Mesh.ts [hunk 2]
```diff
@@ -477,20 +505,21 @@ export class Mesh extends Container
             this.geometry.dispose();
         }
 
-        this.geometry = null;
+        (this as any).geometry = null;
```
Comment (ivanpopelyshev) [inline packages/mesh/src/Mesh.ts]: @eXponenta changed geometry to private field + readonly prop, but for `destroy` both ways are fine and I want to underline that by approving this PR :)

File: packages/mesh/src/MeshGeometry.ts [hunk 3]
```diff
@@ -20,12 +20,17 @@ import { Buffer, Geometry } from '@pixi/core';
...
+    // @ts-ignore
+    private _updateId: number;
+
     /**
      * @param {Float32Array|number[]} vertices - Positional data on geometry.
      * @param {Float32Array|number[]} uvs - Texture UVs.
      * @param {Uint16Array|number[]} index - IndexBuffer
      */
-    constructor(vertices, uvs, index)
+    constructor(vertices: IArrayBuffer, uvs: IArrayBuffer, index: IArrayBuffer)
```
Comment (ivanpopelyshev) [inline packages/mesh/src/MeshGeometry.ts]: yay! That's why I created IArrayBuffer.

**Comment by Zyie** [comment]
@ivanpopelyshev I do have a question for this pr. 

Inside `MeshGeometry` there is `_updateId`. I couldn't find any reference to it so is it needed?

**Pull request merged**